tests/unit/test_basics.py

```diff
@@ -107,33 +134,115 @@ def assert_user_state_empty():
         self.assertEqual(self.block.show_question_header, False)
         self.assertEqual(self.block.item_background_color, "cornflowerblue")
         self.assertEqual(self.block.item_text_color, "coral")
         self.assertEqual(self.block.weight, 5)
-        self.assertEqual(self.block.data, {'foo': 1})
+        self.assertEqual(self.block.max_items_per_zone, None)
+        self.assertEqual(self.block.data, {'foo': 1, 'items': []})
+
+    def test_studio_submit_assessment(self):
+        def update_submission(submission):
+            submission.update({
+                'mode': DragAndDropBlock.ASSESSMENT_MODE,
+                'max_items_per_zone': 4,
+                'show_problem_header': True,
+                'show_title': True,
+                'max_attempts': 12,
+                'item_text_color': 'red',
+                'data': {'foo': 2, 'items': [{'zone': '1', 'title': 'qwe'}]},
+            })
+
+        body = self._make_submission(update_submission)
+        res = self.call_handler('studio_submit', body)
+        self.assertEqual(res, {'result': 'success'})
+
+        self.assertEqual(self.block.show_title, True)
+        self.assertEqual(self.block.mode, DragAndDropBlock.ASSESSMENT_MODE)
+        self.assertEqual(self.block.max_attempts, 12)
+        self.assertEqual(self.block.display_name, "Test Drag & Drop")
+        self.assertEqual(self.block.question_text, "Problem Drag & Drop")
+        self.assertEqual(self.block.show_question_header, True)
+        self.assertEqual(self.block.item_background_color, "cornflowerblue")
+        self.assertEqual(self.block.item_text_color, "red")
+        self.assertEqual(self.block.weight, 5)
+        self.assertEqual(self.block.max_items_per_zone, 4)
+        self.assertEqual(self.block.data, {'foo': 2, 'items': [{'zone': '1', 'title': 'qwe'}]})
+
+    def test_studio_submit_empty_max_items(self):
+        def update_submission(submission):
+            submission['max_items_per_zone'] = ''
+
+        body = self._make_submission(update_submission)
+        res = self.call_handler('studio_submit', body)
+        self.assertEqual(res, {'result': 'success'})
+
+        self.assertIsNone(self.block.max_items_per_zone)
+
+    @ddt.data(-1, -5, -100, -1e15)
+    def test_studio_submit_max_items_negative(self, max_items_per_zone):
+        def modify_submission(submission):
+            submission['max_items_per_zone'] = max_items_per_zone
+
+        res = self.call_handler('studio_submit', self._make_submission(modify_submission))
+        self.assertEqual(res['result'], 'failure')
+        self.assertEqual(len(res['messages']), 1)
+        self.assertIn('should be positive', res['messages'][0])
+
+    def test_studio_submit_max_items_validation(self):
+        def submission_success(submission):
+            submission['max_items_per_zone'] = 1
+            submission['data']['items'] = [
+                {'zones': ['1'], 'title': 'item 1'}, {'zones': ['2'], 'title': 'item 2'}
```

Comment:
@e-kolpakov For consistency with the other submission modification methods, should zones be called `'Zone 1'` and `'Zone 2'` here as well?